fbw-a380x/src/systems/instruments/src/MFD/pages/common/DataEntryFormats.tsx

```diff
@@ -1412,7 +1419,7 @@ export class SquawkFormat implements DataEntryFormat<number> {
     }
 
     const nbr = Number(input);
-    if (!Number.isNaN(nbr) && nbr <= this.maxValue && nbr >= this.minValue) {
+    if (!Number.isNaN(nbr) && nbr <= this.maxValue && nbr >= this.minValue && /^[0-7]{4}$/.test(input)) {
```

Comment:
The min/max checks are redundant with the regex, but not harmful.

Reply:
cleaned that up a little bit